### PR TITLE
New package:  w2c2 WebAssembly to C transpiler

### PR DESCRIPTION
**Pull request by SamuraiCrow**
This recipe clears the way for WASI formatted WebAssembly modules to be compiled to Haiku packages from HaikuPorter.

### CODE REVIEW

**Comment by SamuraiCrow** [comment]
@OscarL , thanks for the "head's up".  I noticed that about the PR title after I posted it.  Also, thanks for the gentile critiques.  This commit should have fixed both redudant lines of code.

File: dev-lang/w2c2/w2c2-0.1.recipe [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+COPYRIGHT="2021 Bastian Mueller"
+LICENSE="MIT"
+REVISION="1"
+srcGitRev="cd90cd995501d83ce41f57bfb11b1a52de092d20"
+SOURCE_URI="https://github.com/turbolent/w2c2/archive/$srcGitRev.zip"
+CHECKSUM_SHA256="8bc9699771b03f92237980e24805f74e4b866d57130089e968c038d364b0de2a"
+SOURCE_FILENAME="w2c2-$srcGitRev.zip"
+SOURCE_DIR="w2c2-$srcGitRev"
+
+ARCHITECTURES="ALL"
```
Comment (korli) [inline dev-lang/w2c2/w2c2-0.1.recipe:23]:
```suggestion
ARCHITECTURES="all"
```

Comment (korli) [reply dev-lang/w2c2/w2c2-0.1.recipe:23]: you think this build on x86_gcc2?

File: dev-lang/w2c2/w2c2-0.1.recipe [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+PROVIDES="
+	w2c2
+	cmd:w2c2
+	"
+REQUIRES="
+	haiku
+	"
+BUILD_REQUIRES="
```
Comment (korli) [inline dev-lang/w2c2/w2c2-0.1.recipe:32]:
```suggestion

BUILD_REQUIRES="
```

File: dev-lang/w2c2/w2c2-0.1.recipe [hunk 3]
```diff
@@ -20,18 +20,21 @@ CHECKSUM_SHA256="8bc9699771b03f92237980e24805f74e4b866d57130089e968c038d364b0de2
...
 SOURCE_DIR="w2c2-$srcGitRev"
 
-ARCHITECTURES="ALL"
+ARCHITECTURES="all"
 
 PROVIDES="
 	w2c2
 	cmd:w2c2
 	"
+
```
Comment (korli) [inline dev-lang/w2c2/w2c2-0.1.recipe:29]:
```suggestion
```

File: dev-lang/w2c2/w2c2-0.1.recipe [hunk 4]
```diff
@@ -20,18 +20,21 @@ CHECKSUM_SHA256="8bc9699771b03f92237980e24805f74e4b866d57130089e968c038d364b0de2
...
 REQUIRES="
 	haiku
 	"
+
 BUILD_REQUIRES="
 	haiku_devel
 	"
+
```
Comment (korli) [inline dev-lang/w2c2/w2c2-0.1.recipe:37]:
```suggestion
```

**Comment by Begasus** [comment]
As this is a commit checkout (srcGitRev), maybe rename the recipe to match the commit date? Like: `w2c2-0.1~19122022.recipe`

**Comment by SamuraiCrow** [comment]
@Begasus Agreed and done!

**Comment by korli** [comment]
usually the format is 20221219 because it is treated as a string, and compared alphabetically.

**Comment by SamuraiCrow** [comment]
@korli Done!

**Comment by Begasus** [comment]
> usually the format is 20221219 because it is treated as a string, and compared alphabetically.

My bad, guess I'm still to used on the European format?

**Pull request merged**